exercises/book-store/src/example/java/BookStore.java

```diff
@@ -0,0 +1,62 @@
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+public class BookStore{
+	public static double CalculateTotalCost(ArrayList<Integer> books ){
+		return CalculateTotalCost(books,0);
+	}
+
+	public static double CalculateTotalCost (ArrayList<Integer> books,double priceSoFar ){
+		double minPrice = Double.MAX_VALUE;
+
+		if(books.size() == 0)
+			return priceSoFar;
+
+		ArrayList<Integer> groups = (ArrayList<Integer>) books.stream().distinct().collect(Collectors.toList());
+
+		
+		double price = 0;
+
+		for(int i = 0;i<groups.size();i++){
+			books.remove(groups.get(i));	
+		}
+		
+		
+		try {
+			price = CalculateTotalCost(books,priceSoFar + CostPerGroup(groups.size()));
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+		minPrice = Math.min(minPrice, price);
+		return minPrice;
+
+
+	}
+
+	private static double CostPerGroup(int groupSize) throws Exception{
+		double discountPercentage;
+		switch (groupSize)
+		{
+		case 1:
+			discountPercentage = 0;
+			break;
+		case 2:
+			discountPercentage = 5;
+			break;
+		case 3:
+			discountPercentage = 10;
+			break;
+		case 4:
+			discountPercentage = 20;
+			break;
+		case 5:
+			discountPercentage = 25;
+			break;
```

Comment:
``` java 
8 * (groupSize - 1) * 0.05d;
```

Are you sure that this work ? Or maybe I'm missing something ?
My method looks like this : 
```java
private double CostPerGroup(int groupSize) throws Exception{
	if(groupSize > maxGroupSize || groupSize <= 0 )
		throw new Exception("Invalide group size : " + groupSize );

	double discountForOneBook = bookPrice * (groupSize - 1) * discountIncrement;
	return (bookPrice * groupSize ) - (groupSize * discountForOneBook);
}
```
But by running gradle for the test, I have 13 tests completed, 5 failed  : 
```
BookStoreTest > TwoCopiesOfEachBook FAILED
    java.lang.AssertionError: expected:<60.0> but was:<64.0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:553)
        at org.junit.Assert.assertEquals(Assert.java:683)
        at BookStoreTest.TwoCopiesOfEachBook(BookStoreTest.java:106)


BookStoreTest > TwoEachOfFirst4BooksAnd1CopyEachOfRest FAILED
    java.lang.AssertionError: expected:<55.6> but was:<59.2>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:553)
        at org.junit.Assert.assertEquals(Assert.java:683)
        at BookStoreTest.TwoEachOfFirst4BooksAnd1CopyEachOfRest(BookStoreTest.java:97)

BookStoreTest > ThreeEachOFirst2BooksAnd2EachOfRemainingBooks FAILED
    java.lang.AssertionError: expected:<75.2> but was:<79.2>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:553)
        at org.junit.Assert.assertEquals(Assert.java:683)
        at BookStoreTest.ThreeEachOFirst2BooksAnd2EachOfRemainingBooks(BookStoreTest.java:124)

BookStoreTest > ThreeCopiesOfFirstBookAnd2EachOfRemaining FAILED
    java.lang.AssertionError: expected:<68.0> but was:<72.0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:553)
        at org.junit.Assert.assertEquals(Assert.java:683)
        at BookStoreTest.ThreeCopiesOfFirstBookAnd2EachOfRemaining(BookStoreTest.java:115)

BookStoreTest > TwoGroupsOfFourIsCheaperThanGroupOfFivePlusGroupOfThree FAILED
    java.lang.AssertionError: expected:<51.2> but was:<53.6>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:553)
        at org.junit.Assert.assertEquals(Assert.java:683)
        at BookStoreTest.TwoGroupsOfFourIsCheaperThanGroupOfFivePlusGroupOfThree(BookStoreTest.java:79)
```
So, I prefer to upload this modifications and don't push now.
[BookStore_Java_Twisti.tar.gz](https://github.com/exercism/xjava/files/847809/BookStore_Java_Twisti.tar.gz)